src/loaders/FileLoader.js

```diff
@@ -55,225 +53,178 @@ class FileLoader extends Loader {
 
 		}
 
-		// Check for data: URI
-		const dataUriRegex = /^data:(.*?)(;base64)?,(.*)$/;
-		const dataUriRegexResult = url.match( dataUriRegex );
-		let request;
-
-		// Safari can not handle Data URIs through XMLHttpRequest so process manually
-		if ( dataUriRegexResult ) {
-
-			const mimeType = dataUriRegexResult[ 1 ];
-			const isBase64 = !! dataUriRegexResult[ 2 ];
-
-			let data = dataUriRegexResult[ 3 ];
-			data = decodeURIComponent( data );
-
-			if ( isBase64 ) data = atob( data );
+		// Initialise array for duplicate requests
+		loading[ url ] = [];
 
-			try {
-
-				let response;
-				const responseType = ( this.responseType || '' ).toLowerCase();
-
-				switch ( responseType ) {
-
-					case 'arraybuffer':
-					case 'blob':
+		loading[ url ].push( {
+			onLoad: onLoad,
+			onProgress: onProgress,
+			onError: onError,
+		} );
 
-						const view = new Uint8Array( data.length );
+		// TODO-DefinitelyMaybe: Confirm if Safari can handle Data URIs through fetch
```

Comment:
Right. Confirmation on Safari and we're good to go?

Reply:
Tried a few examples in a console on Safari 15, which worked fine!

If the changed is merged I can test this with a few glTF files with embedded Data URIs too.

Reply:
@DefinitelyMaybe I guess the TODO can be removed then?

Reply:
ah, yes, indeed 😄